_docs/setup/kubernetes/upgrading-istio.md

```diff
@@ -12,36 +12,47 @@ for the Kubernetes environment.
 
 ## Overview
 
-This guide describes how to upgrade an existing Istio deployment (including
-both control plane and sidecar proxy) to a new release of Istio. The upgrade
-process could involve new binaries as well as other changes like configuration
-and API schemas. The upgrade process may involve some service downtime.
+This guide describes how to upgrade an existing Istio deployment (including both control plane and sidecar proxy) to a new release of Istio. The upgrade process could involve new binaries as well as other changes like configuration and API schemas. The upgrade process may involve some service downtime.   To minimize the service downtime, please ensure your istio control plane components and your application are highly available with multiple replicas.
 
 ## Application setup
 
-In the following steps, we assume that the Istio components are installed and
-upgraded in the same namespace ISTIO\_NAMESPACE.
+In the following steps, we assume that the Istio components are installed and upgraded in the the `istio-system` namespace.
 
 ## Tasks
 
 ### Control plane upgrade
 
-The Istio control plane components include: Citadel, Ingress, Pilot, Mixer, and
+The Istio control plane components include: Citadel, Ingressgateway, Egressgateway, Pilot, Policy, Telemtry and
 Sidecar injector. We can use Kubernetes’ rolling update mechanism to upgrade the
-control plane components. It can be done by simply applying the new version
-yaml file directly, e.g.
+control plane components.  
+
+First, generate the desired istio control plane yaml file, e.g.
 
 ```command
-$ kubectl apply -f istio.yaml (or istio-auth.yaml)
+helm template  --set global.proxy.image=proxy \
```

Comment:
Should not here be `global.proxy.image=proxyv2`?

Reply:
no, because we want folks to use proxy for backwards compatible.